test/WebJobs.Script.Tests.Integration/Host/HostIdValidatorTests.cs

```diff
@@ -127,10 +127,13 @@ public async Task ValidateHostIdUsageAsync_Collision_WarningLevel_Logs()
             await _hostIdValidator.ValidateHostIdUsageAsync(_testHostId);
 
             var logs = _loggerProvider.GetAllLogMessages();
-            var log = logs.Single();
+            var log = logs[0];
             Assert.Equal(LogLevel.Warning, log.Level);
             Assert.Equal(string.Format(Resources.HostIdCollisionFormat, _testHostId), log.FormattedMessage);
 
+            var diagLog = logs[1];
+            Assert.Contains("collision", diagLog.FormattedMessage);
```

Comment:
Is there a reason we're not validating against the whole error message, including the host ID?

Reply:
We can validate against that too. It is the same message as in the log above; I will add that validation.